source/common/http/http2/protocol_constraints.h

```diff
@@ -9,12 +9,27 @@
 #include "source/common/http/http2/codec_stats.h"
 #include "source/common/http/status.h"
 
+#ifdef ENVOY_NGHTTP2
 #include "nghttp2/nghttp2.h"
+#endif
 
 namespace Envoy {
 namespace Http {
 namespace Http2 {
 
+enum FrameType {
```

Comment:
```suggestion
enum class FrameType {
```

This will provide useful context wherever these values are used. I.e. FrameType::PRIORITY is more readable then just plain PRIORITY.